### ISSUE TITLE
Improve error reporting of PeerIdTransport

### ISSUE THREAD

**Issue by tomaka** [libp2p/rust-libp2p#322]
Dialing a `/p2p/...` address returns a `Future`.
However the error produced by this future is always the same: "failed to find multiaddress".

The error message should at least be different if there was no multiaddress available at all, or if there was only one, so that issues are easier to debug.

**Comment by tomaka** [comment]
No longer relevant after #723
